lib/elixir/lib/macro.ex

```diff
@@ -131,7 +131,47 @@ defmodule Macro do
 
   @typep output_expr :: {output_expr | atom, metadata, atom | [output]}
 
-  @typedoc "A keyword list of AST metadata"
+  @typedoc """
+  A keyword list of AST metadata.
+
+  The metadata in Elixir AST is a keyword list of values. Any key can be used
+  and different parts of the compiler may use different keys. For example,
+  the AST received by a macro will always include the `:line` annotation,
+  while the AST emitted by `quote/2` only has the `:line` annotation if the
+  `:line` option is provided.
+
+  The following metadata keys are public:
+
+    * `:column` - The column number of the AST node. It is included only
+      in the AST generated by `Code.string_to_quoted/2` with the option `columns: true`.
+    * `:context` - Defines the context in which the AST was generated.
+      For example, `quote/2` will include the module calling `quote/2`
+      as the context. This is often used to distinguish code from code generated
```

Comment:
This is it not clear. Should we say "This is often used to distinguish **regular** code from code generated by a macro or by `quote/2`" ?